lib3h/src/transport/mod.rs

```diff
@@ -0,0 +1,45 @@
+//! common types and traits for working with Transport instances
+
+mod error;
+
+/// a connection identifier
+pub type TransportId = String;
+pub type TransportIdRef = str;
+
+pub use self::error::{TransportError, TransportResult};
+
+/// type name for a bool indicating if work was done during a `poll()`
+pub type DidWork = bool;
+
+/// events that can be generated during a connection `poll()`
+#[derive(Debug, PartialEq, Clone)]
+pub enum TransportEvent {
+    TransportError(TransportId, TransportError),
+    Connect(TransportId),
+    Message(TransportId, Vec<u8>),
+    Close(TransportId),
+}
+
+/// represents a pool of connections to remote nodes
```

Comment:
This right here implies calling transport a connection instead would be a bit misleading.